ext/mbstring/tests/mb_strpos.phpt

```diff
@@ -70,16 +70,31 @@ echo "== NO OFFSET AND ENCODING PARAMETER ==\n";
 
 mb_internal_encoding('EUC-JP')  or print("mb_internal_encoding() failed\n");
 
-print  mb_strpos($euc_jp, '���ܸ�') . "\n";
+print  mb_strpos($euc_jp, "\xC6\xFC\xCB\xDC\xB8\xEC") . "\n";
 print  mb_strpos($euc_jp, '0') . "\n";
 print  mb_strpos($euc_jp, 3) . "\n";
 print  mb_strpos($euc_jp, 0) . "\n";
 
-$r = mb_strpos($euc_jp, '�ڹ��');
+$r = mb_strpos($euc_jp, "\xB4\xDA\xB9\xF1\xB8\xEC");
 ($r === FALSE) ? print "OK_STR\n"     : print "NG_STR\n";
 $r = mb_strpos($euc_jp, "\n");
 ($r === FALSE) ? print "OK_NEWLINE\n" : print "NG_NEWLINE\n";
 
+echo "== INVALID STRINGS ==\n";
+
+// Previously, mb_strpos would internally convert invalid byte sequences to '?'
+// BEFORE performing search
+// (This was regardless of the setting of mb_substitute_char)
+// So invalid byte sequences would match '?', both from haystack to needle
+// and needle to haystack
+
+var_dump(mb_stripos("abc??", "\xFF", 0, "UTF-8")); // should be false
+var_dump(mb_stripos("abc\xFF", "?", 0, "UTF-8")); // should be false
+
+// However, invalid byte sequences can still match invalid byte sequences:
+var_dump(mb_stripos("abc\x80\x80", "\xFF", 0, "UTF-8"));
+var_dump(mb_stripos("abc\xFF", "c\x80", 0, "UTF-8"));
+
```

Comment:
This file is testing of `mb_strpos`, but `mb_stripos` testing. Is not about `mb_strpos` ?

Reply:
Fixed. Thanks!